src/input/element-input.js

```diff
@@ -948,6 +948,11 @@ class ElementInput {
         for (let i = 0, len = this._elements.length; i < len; i++) {
             const element = this._elements[i];
 
+            // check if any of the layers this element renders to is being rendered by the camera
+            if (!element.layers.some(v => camera.layers.indexOf(v) !== -1)) {
```

Comment:
Yes, I totally agree :/ To be honest I've been trying to figure out a better solution but couldn't find anything. I'm sending the PR to see if anyone has any ideas.

Reply:
the easiest way would be to add camera.layers to a Set. Either inside the Camera class .. when `set layers` is called, it copies data to an array .. it should also build a Set (few classes in the engine do this already, for example in LayerComposition). Then `camera.layers.indexOf(v)` would be `camera.layersSet.has(v)` or similar.

You could also have global `const _tempSet = new Set();` and populate it inside `_getTargetElement` function (outside of the loop) - I assume this is called once a frame only?

Reply:
that you still only bring O(n^2) to O(n) .. going to O(1) would require bitflags on both side .. which would not be super easy. But perhaps O(n) here is enough, especially since the n is usually pretty small here.